### ISSUE TITLE
Unicode chars incorrectly read with driver 2.0

### ISSUE THREAD

**Issue by RenatoMartins-tomtom** [aws/amazon-redshift-odbc-driver#6]
## Driver version
2.00.00.01

## Redshift version
Redshift 8.0.2
PostgreSQL 8.0.2 on i686-pc-linux-gnu, compiled by GCC gcc (GCC) 3.4.2 20041017 (Red Hat 3.4.2-6.fc3), Redshift 1.0.44126

## Client Operating System
Win 10 21H2 (19044.2251)

## Table schema
Any varchar column

## Problem description

1. Expected behaviour: Non-printable Unicode characters shouldn't be shown.
2. Actual behaviour: Non-printable Unicode characters are shown
4. Error message/stack trace: N/A
5. Any other details that can be helpful: Problem is easy to reproduce. Store a concat of CHR(8203) and any string. Unicode 8203 is "zero width space". JDBC driver and ODBC v1 properly output this as empty space to clients.

## ODBC trace logs
N/A

## Reproduction code
Any table with a varchar column. Insert data containing char(8203), and try to consume in clients. In my case, the client is PowerBI. Messy values with the new driver
![image](https://user-images.githubusercontent.com/75678221/216049438-71f1cf7b-e3b2-491d-acb5-32c86d56bc8a.png)

old driver, same dataset, showing the right values
![image](https://user-images.githubusercontent.com/75678221/216051458-3811a1fc-134c-49e9-b61d-d13fed45b232.png)

**Comment by Brooke-white** [comment]
Hi @RenatoMartins-tomtom, thank you for reporting this issue. I've forwarded this to our ODBC development team for investigation and will update when available.

**Comment by vahid110** [comment]
@RenatoMartins-tomtom Thanks for reaching out. 

A unicode conversion fix has been included in the upcoming release. I just added chr(8203) to some fields (including VARCHAR as well as column names) in one of the manual tests and things worked. 

The change will be included in the upcoming release.

```
#!/usr/bin/env python
# coding: utf-8
''' This test aims to verify unicode conversion of cell values and column names.
    It verifies column names using a VIEW
'''
import pyodbc, sys,traceback
from array import *
try:
    T = [['我的数@据1111','我的数@@据1111','我的数@@@据1111'], 
        ['我的数`据2222','我的数``据2222','我的数```据2222'], 
        ['我的数†据3333','我的数††据3333','我的数†††据3333'], 
        ['我的数µ据4444','我的数µµ据4444','我的数µµµ据4444' + chr(8203)]]

    C = ["col编号","col名称",chr(8203) + "col地i" + chr(8203) + "址" + chr(8203)]

    cnxn = pyodbc.connect('Driver={Amazon Redshift ODBC Driver (x64)};Server=MYSERVER;Port=5439;Database=dev;UID=auser;PWD={Pdtst;123}')
    cursor = cnxn.cursor()

    cursor.execute('START TRANSACTION')
    cursor.execute('DROP TABLE IF EXISTS testtb001 CASCADE')
    cursor.commit()
    cursor.execute('''
    CREATE TABLE IF NOT EXISTS testtb001 
    (
    p_partkey     INTEGER NOT NULL,
    p_cx      VARCHAR(100) NOT NULL,
    p_c0      VARCHAR(100) NOT NULL,
    p_c1      VARCHAR(100) NOT NULL,
    p_c2      VARCHAR(100) NOT NULL
    );'''
    )
    cursor.execute('COMMIT')
    #insert unicode values
    cursor.execute(f"INSERT into  testtb001 VALUES(111,'t001','{T[0][0]}','{T[0][1]}','{T[0][2]}');")
    cursor.execute(f"INSERT into  testtb001 VALUES(222,'t002','{T[1][0]}','{T[1][1]}','{T[1][2]}');")
    cursor.execute(f"INSERT into  testtb001 VALUES(333,'t003','{T[2][0]}','{T[2][1]}','{T[2][2]}');")
    cursor.execute(f"INSERT into  testtb001 VALUES(444,'t004','{T[3][0]}','{T[3][1]}','{T[3][2]}');")
    #create a view with unicode column names
    t = '''CREATE OR REPLACE VIEW testview02 ({0},{1},{2}) AS select case when p_cx is null then '未知' else p_cx end, p_partkey , p_c0 from testtb001;
    '''.format(C[0], C[1], C[2])
    cursor.execute(t)
    cursor.commit()
    cursor.execute('SELECT * FROM testtb001')

    rows = cursor.fetchall()

    print('test table\'s unicode cell values....', end = '')
    for i in range(len(rows)):
        row = rows[i]
        #print(row)
        assert row.p_c0.encode('utf-8') == T[i][0].encode('utf-8'), f"{i},{j}: row fetch failed! Expected {T[i][0].encode('utf-8')}, Actual {row.p_c0.encode('utf-8')}"
        assert row.p_c1.encode('utf-8') == T[i][1].encode('utf-8'), f"{i},{j}: row fetch failed! Expected {T[i][1].encode('utf-8')}, Actual {row.p_c1.encode('utf-8')}"
        assert row.p_c2.encode('utf-8') == T[i][2].encode('utf-8'), f"{i},{j}: row fetch failed! Expected {T[i][2].encode('utf-8')}, Actual {row.p_c2.encode('utf-8')}"
    print('pass')

    print('test view\'s unicode cell values...', end = '')
    cursor.execute('SELECT * FROM testview02')
    rows = cursor.fetchall()
    for i in range(len(rows)):
        row = rows[i]
        #print(row)
        assert row[2].encode('utf-8') == T[i][0].encode('utf-8'), f"{i}: view row fetch failed!Expected {T[i][0].encode('utf-8')}, Actual {row[2].encode('utf-8')}"
    print('pass')

    print('test view\'s unicode column name....', end = '')
    columns = [column[0] for column in cursor.description]
    for i in range(len(C)):
        # print(columns[i], len(columns[i]))
        assert columns[i].encode('utf-8') == C[i].encode('utf-8'), f"{i}: View Column error: Expected {C[i].encode('utf-8')}, Actual {columns[i].encode('utf-8')}"
    print('pass')
except AssertionError:
    _,msg, tb = sys.exc_info()
    traceback.print_tb(tb) # Fixed format
    tb_info = traceback.extract_tb(tb)
    filename, line, func, text = tb_info[-1]
    print(f'Error: {msg}')
    sys.exit(1)
finally:
    cnxn.close()

sys.exit(0)
```

**Comment by RenatoMartins-tomtom** [comment]
Hi Vahid.

Thanks for the reply. I didn’t see this comment in the issue, oddly enough..

Another question is I see you don’t publish releases in a standard way in the driver repo. Do you have an ETA for this 2.0.0.2 version? How can I know a new release came? By looking at the readme edits, or the aws doc here https://docs.aws.amazon.com/redshift/latest/mgmt/odbc20-install-config-win.html ?

@***@***.******@***.***>, sorry for looping you on this also. Still puzzled as of how to know about new versions, as the repo doesn’t follow a standard way of releases in there, as they are dumped into an S3 bucket.

Regards,
> Renato Martins  /  Cloud Solutions Architect CCoE /  TomTom Service Platform / ***@***.******@***.***>  /  www.tomtom.com<http://www.tomtom.com/>

**Comment by vahid110** [comment]
Hi @RenatoMartins-tomtom ,
After posting a message I realised that I have posted it after following some standard regulations. So, please stay tuned. I'll be back to you shortly.

**Comment by vahid110** [comment]
Hi @RenatoMartins-tomtom 
And sorry for the inconvinience caused by the long delay. 
Kindly aloow me to be more specific about the publish and installation guidelines once I discussed this internally.

At this moment, I can say that we fortunately have made some progress in this issue and a release will be published soon.

By the way, 
we are considering to upgrade our Windows build requirements to Visual Studio 2022 and C++17. Appreciate your comments on that.

**Comment by vahid110** [comment]
Closing the issue as a fix was provided in v2.0.0.3.
